clients/admin-ui/src/features/system/form.ts

```diff
@@ -105,6 +105,7 @@ export const transformFormValuesToSystem = (
         ...payload.meta.vendor,
         id: formValues.meta.vendor.id,
       };
```

Comment:
Should we remove this?

Reply:
yes, we shouldn't consult `meta` at all IMO. that should simplify things quite a bit, for instance this func does a lot just to make sure the `meta` value is ok, when instead, we should be able to do 

```typescript
const payload : System = {
  ...
  vendor_id: formValues.vendor_id;
}
```
and we shouldn't need the whole `if(features.plus)` block